pkg/resources/resource_monitor.go

```diff
@@ -283,13 +277,54 @@ func extractTriggerInts(s sql.NullString) ([]int, error) {
 	return out, nil
 }
 
+// inSlice returns true if n is in string slice h otherwise false.
+func inSlice(n string, h []string) bool {
+	for _, v := range h {
+		if v == n {
+			return true
+		}
+	}
+	return false
+}
+
+// convertTerraformSetToStringSlice turns a terraform set into a string slice.
+func convertTerraformSetToStringSlice(i interface{}) []string {
+	s := make([]string, len(i.(*schema.Set).List()))
+	for x, w := range i.(*schema.Set).List() {
+		s[x] = w.(string)
+	}
+	return s
+}
+
+// compareTerraformSets compares two terraform sets and returns a string slice of all values in the first set that is
+// not in the second set.
+func compareTerraformSets(firstSet interface{}, secondSet interface{}) []string {
```

Comment:
firstly, helper functions like this should be grouped with the other helper functions. they are spread out a bit now, but i think we should start putting them together in the helpers package https://github.com/Snowflake-Labs/terraform-provider-snowflake/tree/main/pkg/helpers.

second, this function is unclear in arguments. The input arguments are both generic interface type, but the result is a slice of string. I think if you want to deal with slices, then deal with slices. Don't do this extra conversion behind the scenes. 

Finally, the function is unclear in name. What is it exactly that you are comparing? I dont think you are actually comparing anything, what you are doing is finding either the union, intersection or difference.